platform/ios/display_server_ios.mm

```diff
@@ -109,18 +109,39 @@
 #endif
 	if (rendering_context) {
 		if (rendering_context->initialize() != OK) {
-			memdelete(rendering_context);
-			rendering_context = nullptr;
+			bool failed = true;
+#if defined(VULKAN_ENABLED)
+			bool fallback_to_vulkan = GLOBAL_GET("rendering/rendering_device/fallback_to_vulkan");
+			if (failed && fallback_to_vulkan && rendering_driver != "vulkan") {
+				layer = [AppDelegate.viewController.godotView initializeRenderingForDriver:@"vulkan"];
+				if (!layer) {
+					ERR_FAIL_MSG("Failed to create iOS Vulkan rendering layer.");
+				}
+				wpd.vulkan.layer_ptr = (CAMetalLayer *const *)&layer;
+				rendering_context = memnew(RenderingContextDriverVulkanIOS);
```

Comment:
This will overwrite the previous value of `rendering_context` unless `memdelete` is restored to line 112